pkg/api/pluginproxy/ds_proxy_test.go

```diff
@@ -895,3 +895,40 @@ func runDatasourceAuthTest(t *testing.T, test *testCase) {
 
 	test.checkReq(req)
 }
+
+func Test_PathCheck(t *testing.T) {
+	// Ensure that we test routes appropriately. This test reproduces a historical bug where two routes were defined with different role requirements but the same method and the more privileged route was tested first. Here we ensure auth checks are applied based on the correct route, not just the method.
+	plugin := &plugins.DataSourcePlugin{
```

Comment:
Another interesting case (if even possible) would be if the routes had the same path as well.

Reply:
I think two route entries with the same method + same path are indistinguishable and so that'd be considered a schema bug I believe. We could probably validate that during plugin loading, but it seems out of scope for this PR.